src/dune_pkg/opam_repo.mli

```diff
@@ -21,8 +21,12 @@ val of_opam_repo_dir_path
   -> Path.t
   -> t
 
-(** [of_git_repo git source] loads the data through git *)
-val of_git_repo : repo_id:Repository_id.t option -> source:string -> t Fiber.t
+(** [of_git_repo git ~repo_id ~update ~source] loads the opam repository from git. *)
+val of_git_repo
+  :  repo_id:Repository_id.t option
+  -> update:bool
+  -> source:string
```

Comment:
can you document what `source` means or use a type other than `string`

Reply:
That's meant to be the url or path I believe.

Reply:
Good point. I went with documenting it because besides doing a dummy type that represents "whatever `git remote add` accepts" with very little validation that we can do on our side.

Reply:
I will probably be reworking this in the near future to have a source specification format with repo/branch/tags.